packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt

```diff
@@ -47,3 +57,72 @@ internal fun Map<*, *>.toWritableMap(): WritableNativeMap {
     }
     return map
 }
+
+/**
+ * Casts the ReadableArray to a List that strictly contains non-null objects of type HashMap.
+ * @throws TypeCastException if at least one of the list items is not of type HashMap.
+ */
+internal fun ReadableArray.toHashMapArrayList(): List<HashMap<*, *>> {
+    return this.toKotlinArrayList().toMutableList().mapNotNull {
+        if (it is HashMap<*, *> || it == null) {
+            it as HashMap<*, *>?
+        } else {
+            throw TypeCastException(
+                "Cannot convert ReadableArray to ArrayList of HashMap(s): " +
+                    "$it is not of type HashMap<*,*>",
+            )
+        }
+    }
+}
+
+/**
+ * Recursively converts the ReadableMap to a HashMap which only contains Kotlin stdlib objects,
+ * such as ArrayList, HashMap and the raw types.
+ * @throws IllegalArgumentException if the readable map is malformed (eg. it contains a HashMap
+ * or ArrayList, instead of ReadableMap and ReadableArray respectively).
+ */
+internal fun ReadableMap.toKotlinHashMap(): HashMap<String, Any> {
+    val hashMap: HashMap<String, Any> = this.toHashMap()
+    val iterator: Iterator<*> = hashMap.keys.iterator()
+
+    while (iterator.hasNext()) {
+        val key = iterator.next() as String
+        when (getType(key)) {
+            ReadableType.Null, ReadableType.Boolean, ReadableType.Number, ReadableType.String -> {}
+            ReadableType.Map -> hashMap[key] = (Assertions.assertNotNull(getMap(key)) as ReadableMap).toKotlinHashMap()
+            ReadableType.Array -> hashMap[key] = (Assertions.assertNotNull(getArray(key)) as ReadableArray).toKotlinArrayList()
+            else -> throw IllegalArgumentException("Could not convert object with key: $key.")
+        }
+    }
+
+    return hashMap
+}
+
+/**
+ * Recursively converts the ReadableArray to a ArrayList which only contains Kotlin stdlib objects,
+ * such as ArrayList, HashMap and the raw types.
+ * @throws IllegalArgumentException if the readable array is malformed (eg. it contains a HashMap
+ * or ArrayList, instead of ReadableMap and ReadableArray respectively).
+ */
+internal fun ReadableArray.toKotlinArrayList(): ArrayList<*> {
+    val arrayList = ArrayList<Any?>()
+    for (i in 0 until size()) {
+        // ReadableArray throws a null pointer exception if getMap(i) or getArray(i) returns null
```

Comment:
For JavaOnlyArray and JavaOnlyMap, if they contain stdlib objects, the case is not handled correctly. They can't return null, they throw a null pointer exception when trying to access the map or array, because they expect to find a ReadableMap or a ReadableArray respectively.

Reply:
How it is possible and where it happens? I'm checking the source code of `JavaOnlyArray`, for example.

Here, there is a cast https://github.com/facebook/react-native/blob/66634be13a37779bbafd75f24a88c97cf706cd2d/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/JavaOnlyArray.java#L121 in the `getMap` call.

It should be no problem doing this cast and getting underlying value if `getType` call is made before, because `ReadableType.Map` [guarantees](https://github.com/facebook/react-native/blob/66634be13a37779bbafd75f24a88c97cf706cd2d/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/JavaOnlyArray.java#L147) that there is indeed `ReadableMap` behind this index.